global_helpers/panther_oss_helpers.py

```diff
@@ -162,263 +162,88 @@ def resource_lookup(resource_id: str) -> Dict[str, Any]:
-        ExpressionAttributeValues={":incr": val},
-    )
-
-    # Numeric values are returned as decimal.Decimal
-    return response["Attributes"][_COUNT_COL].to_integral_value()
+    """Global `increment_counter` is DEPRECATED.
+    Instead, use `from panther_detection_helpers.caching import increment_counter`."""
+    return caching.increment_counter(key=key, val=val)
 
 
 def reset_counter(key: str) -> None:
-    """Reset a counter to 0."""
-    kv_table().put_item(Item={"key": key, _COUNT_COL: 0})
+    """Global `reset_counter` is DEPRECATED.
+    Instead, use `from panther_detection_helpers.caching import reset_counter`."""
+    return caching.reset_counter(key=key)
 
 
 def set_key_expiration(key: str, epoch_seconds: int) -> None:
-    """Configure the key to automatically expire at the given time.
-    DynamoDB typically deletes expired items within 48 hours of expiration.
-    Args:
-        key: The name of the counter
-        epoch_seconds: When you want the counter to expire (set to 0 to disable)
-    """
-    if isinstance(epoch_seconds, str):
-        epoch_seconds = float(epoch_seconds)
-    if isinstance(epoch_seconds, float):
-        epoch_seconds = int(epoch_seconds)
-    if not isinstance(epoch_seconds, int):
-        return
-    # if we are given an epoch seconds that is less than
-    # 604800 ( aka seven days ), then add the epoch seconds to
-    # the timestamp of now
-    if epoch_seconds < 604801:
-        epoch_seconds = int(datetime.now().timestamp()) + epoch_seconds
-    kv_table().update_item(
-        Key={"key": key},
-        UpdateExpression="SET expiresAt = :time",
-        ExpressionAttributeValues={":time": epoch_seconds},
-    )
+    """Global `set_key_expiration` is DEPRECATED.
+    Instead, use `from panther_detection_helpers.caching import set_key_expiration`."""
+    return caching.set_key_expiration(key=key, epoch_seconds=epoch_seconds)
 
 
 def put_dictionary(key: str, val: dict, epoch_seconds: int = None):
-    """Overwrite a dictionary under the given key.
-    The value must be JSON serializable, and therefore cannot contain:
-        - Sets
-        - Complex numbers or formulas
-        - Custom objects
-        - Keys that are not strings
-    Args:
-        key: The name of the dictionary
-        val: A Python dictionary
-        epoch_seconds: (Optional) Set string expiration time
-    """
-    if not isinstance(val, (dict, Mapping)):
-        raise Exception("panther_oss_helpers.put_dictionary: value is not a dictionary")
-
-    try:
-        # Serialize 'val' to a JSON string
-        data = json.dumps(val)
-    except TypeError as exc:
-        raise Exception(
-            "panther_oss_helpers.put_dictionary: "
-            "value is a dictionary, but it is not JSON serializable"
-        ) from exc
-
-    # Store the item in DynamoDB
-    kv_table().put_item(Item={"key": key, _DICT_COL: data})
-
-    if epoch_seconds:
-        set_key_expiration(key, epoch_seconds)
```

Comment:
`[out of scope] rant` this should really be one request instead of 2

Reply:
Yes true. We changed it to 1 in the caching module